addons/auth_signup/controllers/main.py

```diff
@@ -25,6 +25,18 @@ def web_login(self, *args, **kw):
             return request.redirect(request.params.get('redirect'))
         return response
 
+    @http.route('/web/redirect', type='http', auth='public', website=True, sitemap=False)
+    def web_redirect(self, token=False, db=False, **kw):
+        if not token or not db:
+            return request.redirect('/web/login')
+        partner = request.env['res.partner'].sudo()._signup_retrieve_partner(token, check_validity=True, raise_exception=False)
+        if not partner or (partner.user_id and partner.user_id.password):
```

Comment:
```suggestion
        if not partner or partner.user_id:
```
or
```suggestion
        if not partner:
```
`password` is always computed to an empty string, so checking for it is useless. Not sure which variant is supposed to occur.